osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -22,14 +22,40 @@
 
 namespace osu.Game.Scoring
 {
+    /// <summary>
+    /// A realm model containing metadata for a single score.
+    /// </summary>
     [ExcludeFromDynamicCompile]
     [MapTo("Score")]
     public class ScoreInfo : RealmObject, IHasGuidPrimaryKey, IHasRealmFiles, ISoftDelete, IEquatable<ScoreInfo>, IScoreInfo
     {
         [PrimaryKey]
         public Guid ID { get; set; }
 
-        public BeatmapInfo BeatmapInfo { get; set; } = null!;
+        /// <summary>
+        /// The <see cref="BeatmapInfo"/> this score was made against.
+        /// </summary>
+        [Ignored]
+        public BeatmapInfo BeatmapInfo
+        {
+            get => BeatmapInfoBacking;
+            set
+            {
+                BeatmapInfoBacking = value;
+                BeatmapHash = value.Hash;
+            }
```

Comment:
```suggestion
            set
            {
                if (value == BeatmapInfoBacking) return;

                BeatmapInfoBacking = value;
                BeatmapHash = value.Hash;
            }
```

Reply:
This changes will break the hash update ([see this](https://github.com/ppy/osu/pull/22539#discussion_r1103612998)).

Reply:
I've pushed changes.